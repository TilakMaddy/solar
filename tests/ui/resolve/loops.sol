function funky() {
    uint i;

    while (i++ < 10) continue;

    do ++i; while (i < 20);
    do {} while (i++ < 30);

    for (;;) break;
    for (; i < 40; i++) continue;
    for (; i++ < 50; ) continue;

    // ---
    // TODO: `Error: Variable declarations can only be used inside blocks.`

    while (a == 0) { //~ ERROR: unresolved symbol 
        uint a = 0;
    }
    a; //~ ERROR: unresolved symbol
    while (b == 0) { //~ ERROR: unresolved symbol
        uint b = 0;
    } 
    b; //~ ERROR: unresolved symbol

    do { uint c; } while (c == 0); //~ ERROR: unresolved symbol
    c; //~ ERROR: unresolved symbol
    do {
        uint d;
    } while (d == 0); //~ ERROR: unresolved symbol
    d; //~ ERROR: unresolved symbol

    for (; false; e++) { uint e; } //~ ERROR: unresolved symbol
    e; //~ ERROR: unresolved symbol
    for (; false; f++) {//~ ERROR: unresolved symbol

        uint f;
    }     f; //~ ERROR: unresolved symbol
    for (uint g; false; g++) {
        g;
    }
    g; //~ ERROR: unresolved symbol
}
