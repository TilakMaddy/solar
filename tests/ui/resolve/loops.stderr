error: variable declaration statements are not allowed as the body of a loop (for, while, do while), meaning they must be inside of a block
  --> ROOT/tests/ui/resolve/loops.sol:LL:CC
   |
LL |     while (a == 0) uint a = 0;
   |                    ^^^^^^^^^^
   |

error: variable declaration statements are not allowed as the body of a loop (for, while, do while), meaning they must be inside of a block
  --> ROOT/tests/ui/resolve/loops.sol:LL:CC
   |
LL |     do uint c; while (c == 0);
   |        ^^^^^^
   |

error: variable declaration statements are not allowed as the body of a loop (for, while, do while), meaning they must be inside of a block
  --> ROOT/tests/ui/resolve/loops.sol:LL:CC
   |
LL |     for (; false; e++) uint e;
   |                        ^^^^^^
   |

error: aborting due to 3 previous errors

